tests/run.go

```diff
@@ -371,38 +381,186 @@ func Run(o *option.Option) {
+			ginkgo.When("--cpuset-cpus is used", func() {
+				ginkgo.It("should set CPUs in which to allow execution as cpu 1 with --cpuset-cpus=1", func() {
+					cpuSet := command.StdoutStr(o.BaseOpt, "run", "--cpuset-cpus", "1",
+						"-w", "/sys/fs/cgroup", defaultImage, "cat", "cpuset.cpus")
+					gomega.Expect(cpuSet).To(gomega.Equal("1"))
+				})
+
+				ginkgo.It("should set CPUs in which to allow execution as cpu 0-1 with --cpuset-cpus=0-1", func() {
+					cpuSet := command.StdoutStr(o.BaseOpt, "run", "--cpuset-cpus", "0-1",
+						"-w", "/sys/fs/cgroup", defaultImage, "cat", "cpuset.cpus")
+					gomega.Expect(cpuSet).To(gomega.Equal("0-1"))
+				})
+
+				ginkgo.It("should set CPUs in which to allow execution as cpu 0-1 with --cpuset-cpus=0,1", func() {
+					cpuSet := command.StdoutStr(o.BaseOpt, "run", "--cpuset-cpus", "0,1",
+						"-w", "/sys/fs/cgroup", defaultImage, "cat", "cpuset.cpus")
+					gomega.Expect(cpuSet).To(gomega.Equal("0-1"))
+				})
+			})
+
+			// The range form (e.g., `--cpuset-mems 0-1`) is not tested because maybe there's only one memory node (i.e., `0`) on the host,
+			// and if that's the case, any number other than 0 would incur an error:
+			// https://man7.org/linux/man-pages/man7/cpuset.7.html#WARNINGS
+			ginkgo.It("should set memory nodes (MEMs) in which to allow execution as memory node 0 with --cpuset-mems=0", func() {
+				cpuMems := command.StdoutStr(o.BaseOpt, "run", "--cpuset-mems", "0",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "cpuset.mems")
+				gomega.Expect(cpuMems).To(gomega.Equal("0"))
+			})
+
+			ginkgo.It("should set the memory limit with --memory", func() {
+				mem := command.StdoutStr(o.BaseOpt, "run", "--memory", "42m",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "memory.max")
+				gomega.Expect(mem).To(gomega.Equal("44040192"))
+			})
+
+			ginkgo.It("should set the memory soft limit with --memory-reservation", func() {
+				mem := command.StdoutStr(o.BaseOpt, "run", "--memory-reservation", "6m",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "memory.low")
+				gomega.Expect(mem).To(gomega.Equal("6291456"))
+			})
+
+			ginkgo.It("should set the amount of memory this container is allowed to swap to disk with --memory-swap", func() {
+				mem := command.StdoutStr(o.BaseOpt, "run", "--memory", "42m", "--memory-swap", "100m",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "memory.max", "memory.swap.max")
+				gomega.Expect(mem).To(gomega.Equal("44040192\n60817408"))
+			})
+
+			// TODO: --memory-swappiness --oom-kill-disable --blkio-weight --cgroupns
+
+			ginkgo.It("should set the container pids limit with --pids-limit", func() {
+				pidsLimit := command.StdoutStr(o.BaseOpt, "run", "--pids-limit", "42",
+					"-w", "/sys/fs/cgroup", defaultImage, "cat", "pids.max")
+				gomega.Expect(pidsLimit).To(gomega.Equal("42"))
+			})
+
+			ginkgo.It("should set the container OOM score with --oom-score-adj", func() {
+				testcases := []string{"100", "1000"}
```

Comment:
After https://github.com/containerd/nerdctl/pull/1622 is merged, the whole thing should be understandable.

```suggestion
				// 100 is the minimum because finch VM runs rootless containerd inside.
				testcases := []string{"100", "1000"}
```